lookout/style/typos/generation.py

```diff
@@ -52,37 +54,61 @@ def __init__(self, **kwargs):
+                                         equal edit distance from the typo to consider as \
+                                         candidates.
+                       max_distance: Maximum edit distance for symspell lookup for candidates.
+                       radius: Maximum edit distance from typo allowed for candidates.
+                       max_corrected_length: Maximum length of prefix in which symspell lookup \
+                                             for typos is conducted.
+                       start_pool_size: Length of data, starting from which multiprocessing is \
+                                        desired.
+                       chunksize: Max size of a chunk for one process during multiprocessing.
         """
-        self.checker = SymSpell(max_dictionary_edit_distance=max_distance,
-                                prefix_length=max_corrected_length)
+        self.set_config(config)
+        self.checker = SymSpell(max_dictionary_edit_distance=self.config["max_distance"],
+                                prefix_length=self.config["max_corrected_length"])
         self.checker.load_dictionary(vocabulary_file)
         self.wv = FastText.load_fasttext_format(embeddings_file).wv
-        self.neighbors_number = neighbors
-        self.edit_candidates_number = edit_candidates
-        self.max_distance = max_distance
-        self.radius = radius
         self.tokens = set(read_vocabulary(vocabulary_file))
         self.frequencies = read_frequencies(frequencies_file)
         self.min_freq = min(self.frequencies.values())
 
+    def set_config(self, config: Optional[Mapping[str, Any]] = None) -> None:
+        """
+        Update candidates generation config.
+
+        :param config: Candidates generation configuration, options:
+                       neighbors_number: Number of neighbors of context and typo embeddings \
+                                         to consider as candidates.
+                       edit_dist_number: Number of the most frequent tokens among tokens on \
+                                         equal edit distance from the typo to consider as \
+                                         candidates.
+                       max_distance: Maximum edit distance for symspell lookup for candidates.
+                       radius: Maximum edit distance from typo allowed for candidates.
+                       max_corrected_length: Maximum length of prefix in which symspell lookup \
+                                             for typos is conducted.
+                       start_pool_size: Length of data, starting from which multiprocessing is \
+                                        desired.
+                       chunksize: Max size of a chunk for one process during multiprocessing.
+        """
+        if config is None:
+            config = {}
+        self.config = merge_dicts(DEFAULT_CORRECTOR_CONFIG["generation"], config)
```

Comment:
Pro tip: you can use [munchify](https://github.com/Infinidat/munch) to work with it like `self.config.chunksize` instead of `self.config["chunk_size"]` while leaving it a `dict`.

Reply:
Not required in this PR, of course